pageserver/src/tenant/storage_layer/delta_layer.rs

```diff
@@ -1539,7 +1575,14 @@ impl<'a> DeltaLayerIterator<'a> {
             .await?;
         let frozen_buf = blobs_buf.buf.freeze();
         for meta in blobs_buf.blobs.iter() {
-            let value = Value::des(&frozen_buf[meta.start..meta.end])?;
+            let slice = &frozen_buf[meta.start..meta.end];
+            let decompressed = vectored_blob_io::decompress(slice, meta.compression_bits).await?;
+            let value = if let Some(decompressed) = &decompressed {
+                Value::des(decompressed)?
+            } else {
+                Value::des(slice)?
+            };
```

Comment:
This is a bit confusing to read. Are we coercing to slices on all of these copypasted?

`Value::des(decompressed.as_deref().unwrap_or(slice))` might work?

Reply:
With the new `BufView` added in 32284e58fc816ba5554a62e7bf5132695bf9478b, we simplifies the control flow.